# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/QuotaParent.h
# Commit: ddb496eb39ab
# Full Hash: ddb496eb39ab8e4a48b7e6924a3a908f30620864
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-09 21:46:09
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Add a dedicated method for resetting storages of a specific client; r=dom-storage-reviewers,jari
#   
#   Client storages can be currently reset by using the optional argument of
#   nsIQuotaManagerService::ResetStoragesForPrincipal. However, supporting
#   resetting storages for both a principal and a client is not ideal for several
# ==============================================================================

diff -r 67a39d84be12 -r ddb496eb39ab dom/quota/QuotaParent.h
--- a/dom/quota/QuotaParent.h	Wed Oct 09 11:59:34 2024 +0000
+++ b/dom/quota/QuotaParent.h	Wed Oct 09 11:59:34 2024 +0000
@@ -114,6 +114,11 @@
       const PrincipalInfo& aPrincipalInfo, const Maybe<Type>& aClientType,
       ShutdownStoragesForOriginResolver&& aResolve) override;
 
+  virtual mozilla::ipc::IPCResult RecvShutdownStoragesForClient(
+      const Maybe<PersistenceType>& aPersistenceType,
+      const PrincipalInfo& aPrincipalInfo, const Type& aClientType,
+      ShutdownStoragesForClientResolver&& aResolve) override;
+
   virtual mozilla::ipc::IPCResult RecvShutdownStorage(
       ShutdownStorageResolver&& aResolver) override;
 